src/main/java/dev/blaauwendraad/masker/json/MaskingState.java

```diff
@@ -49,10 +50,32 @@ public byte[] getMessage() {
         return message;
     }
 
+    public void addReplacementOperation(int startIndex, int endIndex, int maskLength, byte maskByte) {
+        ReplacementOperation replacementOperation = new ReplacementOperation(startIndex, endIndex, maskLength, maskByte);
+        replacementOperations.add(replacementOperation);
+        replacementOperationsTotalDifference += replacementOperation.difference();
+    }
+
+    public List<ReplacementOperation> getReplacementOperations() {
+        return replacementOperations;
+    }
+
+    public int getReplacementOperationsTotalDifference() {
```

Comment:
JavaDoc

Reply:
All the other methods here have no docs whatsoever :D But I'll add one